command/agent/auth/aws/aws.go

```diff
@@ -7,32 +7,54 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"reflect"
+	"sync"
+	"time"
 
+	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/hashicorp/errwrap"
-	cleanhttp "github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-cleanhttp"
 	"github.com/hashicorp/go-hclog"
-	uuid "github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/go-uuid"
 	"github.com/hashicorp/vault/api"
-	awsauth "github.com/hashicorp/vault/builtin/credential/aws"
+	"github.com/hashicorp/vault/builtin/credential/aws"
 	"github.com/hashicorp/vault/command/agent/auth"
 )
 
 const (
 	typeEC2          = "ec2"
 	typeIAM          = "iam"
 	identityEndpoint = "http://169.254.169.254/latest/dynamic/instance-identity"
+
+	/*
+
+		IAM creds can be inferred from instance metadata or the container
+		identity service, and those creds expire at varying intervals with
+		new creds becoming available at likewise varying intervals. Let's
+		default to polling once a minute so all changes can be picked up
+		rather quickly. This is configurable, however.
+
+	*/
+	defaultCredCheckFreqSeconds = 60
```

Comment:
Can we rename this to `defaultCredentialPollInterval`?

Reply:
Oh yes. Thank you Vishal!